pkg/kapp/resourcesmisc/apps_v1_deployment.go

```diff
@@ -59,6 +59,12 @@ func (s AppsV1Deployment) IsDoneApplying() DoneApplyState {
 				return DoneApplyState{Done: true, Successful: false, Message: fmt.Sprintf(
 					"Deployment failed to delete pods: %s (message: %s)", cond.Reason, cond.Message)}
 			}
+
+		case "FailedCreate":
```

Comment:
I just noticed that the switch is on the _type_ of condition, and "FailedDelete" and "FailedCreate" are Reasons for failure for condition type "ReplicaFailure". So instead of "FailedDelete" above and "FailedCreate", we should be checking for `appsv1.DeploymentReplicaFailure`. (More details around status conditions can be found [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status))


It would be great if you could also add a unit test for this in `apps_v1_deployment_test.go`.

Reply:
Thank you so much for the explanation!!! I am working on this.

Reply:
I have added the `appsv1.DeploymentReplicaFailure` instead of failedcreate/failedDelete condition, also I found `apps_v1_deployment_test.go` had no test coverage for `isDoneApplying` function which contains these checks so I added an unit test covering both the checks inside the function ie one for depolyment progressing condition and other for replicafailure.